### ISSUE TITLE
"initialize vision: TLS required" in sing-box version 1.8.0

### ISSUE THREAD

**Issue by yuanyiwei** [SagerNet/sing-box#1334]
### Operating system

Linux

### System version

Debian 12.4

### Installation type

Original sing-box Command Line

### Version

```shell
sing-box version 1.8.0

Environment: go1.21.5 linux/amd64
Tags: with_gvisor,with_quic,with_dhcp,with_wireguard,with_ech,with_utls,with_reality_server,with_acme,with_clash_api
Revision: 11bec79a06268f00e7c5a7d5509245855d6dd522
CGO: disabled
```

### Description

The configuration that works in version 1.7.5 is breaking in version 1.8.0.

I am continuously receiving the following error log: "ERROR [191981011 208ms] inbound/vless[0]: process connection from 127.0.0.1:33333: initialize vision: TLS required."

### Reproduction

```json
{
  "log": {
    "level": "info",
    "timestamp": true
  },
  "inbounds": [
    {
      "type": "vless",
      "listen": "127.0.0.1",
      "listen_port": 666,
      "users": [
        {
          "name": "aaa",
          "uuid": "aaa",
          "flow": "xtls-rprx-vision"
        }
      ],
      "tls": {
        "enabled": true,
        "certificate_path": "/etc/aaa/fullchain.pem",
        "key_path": "/etc/aaa/privkey.pem"
      }
    }
  ]
}
```

just run this config and connect it

### Logs

```shell
+0800 2024-01-07 15:20:18 INFO [470920346 0ms] inbound/vless[0]: inbound connection from 127.0.0.1:62912
+0800 2024-01-07 15:20:18 ERROR [470920346 144ms] inbound/vless[0]: process connection from 127.0.0.1:62912: initialize vision: TLS required
```

### Integrity requirements

- [X] I confirm that I have read the documentation, understand the meaning of all the configuration items I wrote, and did not pile up seemingly useful options or default values.
- [X] I confirm that I have provided the server and client configuration files and process that can be reproduced locally, instead of a complicated client configuration file that has been stripped of sensitive data.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported, instead of depending on remote servers, TUN, graphical interface clients, or other closed-source software.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by zhuzeitou** [comment]
I have the same issue on windows 11 with vless+tcp+xtls-rprx-vision.
A workaround for this is to use utls.

**Comment by zhukovdim** [comment]
I have the same issue (1.8.0 and 1.8.1). On client side utls is enabled.

**Comment by Oliver-Yue** [comment]
> I have the same issue (1.8.0 and 1.8.1). On client side utls is enabled.

The same.

**Comment by nekohasekai** [comment]
Confirmed fix in 1.8.1. Please open a new issue for spam commenters above @zhuzeitou @Oliver-Yue
